phylanx/plugins/matrixops/extract_shape.hpp

```diff
@@ -48,9 +48,12 @@ namespace phylanx { namespace execution_tree { namespace primitives
         primitive_argument_type shape1d(arg_type&& arg) const;
         primitive_argument_type shape2d(arg_type&& arg) const;
 
-        primitive_argument_type shape0d(arg_type&& arg, std::int64_t index) const;
-        primitive_argument_type shape1d(arg_type&& arg, std::int64_t index) const;
-        primitive_argument_type shape2d(arg_type&& arg, std::int64_t index) const;
+        primitive_argument_type shape0d(
+            arg_type&& arg, ir::node_data<std::int64_t> index) const;
+        primitive_argument_type shape1d(
+            arg_type&& arg, ir::node_data<std::int64_t> index) const;
+        primitive_argument_type shape2d(
+            arg_type&& arg, ir::node_data<std::int64_t> index) const;
     };
```

Comment:
These functions are to be called with a single integer as their argument. There is no point for them to be called with a vector or matrix of integers...